spec/lang/step/expressions.md

```diff
@@ -271,16 +271,26 @@ impl<M: Memory> Machine<M> {
 impl<M: Memory> Machine<M> {
     fn eval_place(&mut self, PlaceExpr::Field { root, field }: PlaceExpr) -> Result<(Place<M>, Type)> {
         let (root, ty) = self.eval_place(root)?;
-        let (offset, field_ty) = match ty {
-            Type::Tuple { fields, .. } => fields[field],
-            Type::Union { fields, .. } => fields[field],
+        let ((offset, field_ty), retain_metadata) = match ty {
+            Type::Tuple { sized_fields, .. } if field < sized_fields.len() => (sized_fields[field], false),
+            Type::Tuple { sized_fields, unsized_field, sized_head_layout } if field == sized_fields.len() => {
+                let tail_ty = unsized_field.expect("field projection to non-existing tail");
+                let tail_align = self.compute_align(tail_ty.layout::<M::T>(), root.ptr.metadata);
+                let offset = sized_head_layout.tail_offset(tail_align);
+                ((offset, tail_ty), true)
+            }
+            Type::Union { fields, .. } => (fields[field], false),
             _ => panic!("field projection on non-projectable type"),
```

Comment:
This panic can now also be triggered when the index is too big, I think?

Please have only a single `Type::Tuple` case above and handle all the indexing logic there.